### DISCUSSION TITLE
关于timeout的理解

### DISCUSSION THREAD

**Discussion by He-Sui** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#47] [Q&A] [answered]
```
python3 peer.py -p <peer file> -c <haschunk file> -m <max send> -i <identity> -v <verbosity> [-t <timeout>]
```
在peer的配置信息中对timeout的描述为：
&lt;timeout&gt;: If timeout is not set, you should estimate RTT in the network to set your timeout value. However, if it is set, you should always use this value. A predefined timeout value will be used in the testing.

那么是否是如果传入了timeout的参数，就始终使用这个参数作为timeout的值并且之后不会修改。如果没有传入这个参数，就需要估计RTT并动态更新timeout的值？

**Comment by He-Sui** [comment]
那么在`basic_transfer_test.py`中有这样的描述：
> This test is equivalent to run (except for packet loss):
In shell1:
python3 src/peer.py -p test/tmp2/nodes2.map -c test/tmp2/data1.fragment -m 1 -i 1 -t 60
>
> In shell2:
python3 src/peer.py -p test/tmp2/nodes2.map -c test/tmp2/data2.fragment -m 1 -i 2 -t 60
>
> In shell1:
DOWNLOAD test/tmp2/download_target.chunkhash test/tmp2/download_result.fragment

配置信息是设置timeout为60，并且其中会有丢包，那么至少要等60s才会重新发包，整个跑完至少也需要1分钟，而在其他回答提到可以在1s内运行完？

**Reply by TANSixu** [reply]
这里的重传是由duplicated ACK引起的哈，不是超时引起的。设定60s相当于是屏蔽了超时

**Reply by He-Sui** [reply]
明白了，谢谢！

**Comment by TANSixu** [comment, marked as answer]
Hi，没错，如果传入了timeout则不会再改变。传入timeout参数是因为有basic test不是在模拟器上跑的，估计rtt会有问题。
